ui/raidboss/data/03-hw/trial/sophia-ex.js

```diff
@@ -6,7 +6,409 @@
+      },
+      // We let the storage triggers catch up before calling.
+      delaySeconds: .5,
+      durationSeconds: 12, // Ensuring that forgetful people aren't forgotten.
+      suppressSeconds: 5,
+      alertText: function(data) {
+        // Tethers are ordered with all East tethers first. This *doesn't* mean that the East
+        // or West tethers are themselves in order within their half!
+        // The eight scale entities are listed in the data object, with West at indices 0-3,
+        // under data.scaleSophias.
+        let safeDir = 0;
+        // If there's a side with more tethers, we know for sure that's the safe side.
+        // This will give us the tilt direction for all but the 1/1, 2/2, and 3/3 cases.
+        // The safe side is represented here by whether safeDir is positive or negative.
+        // (West/negative, East/positive.)
+        for (let i = 0; i < data.quasarTethers.length; i++)
+          safeDir += data.scaleSophias.indexOf(data.quasarTethers[i]) < 4 ? -1 : 1;
+        if (safeDir == 0) {
+          // If it's the 1/1, 2/2, or 3/3 case, we sadly don't have enough information.
+          // We have to quit here and wait for the actual cast.
+          return;
+        }
+        return {
+          '2': { en: 'Go East (Hard Tilt)' },
+          '1': { en: 'Go East (Soft Tilt)' },
+          '-2': { en: 'Go West (Hard Tilt)' },
+          '-1': { en: 'Go West (Soft Tilt)' },
+          // Stringified because Javascript doesn't do negative-integer key values.
+        }[safeDir.toString()];
+      },
+    },
+    {
+      // This specifically calls the case where it's 1/1;2/2;3/3 tethers.
+      // The blue Quasar, 19A9, is *alway* on the dangerous side.
+      // The 20/startsUsing log lines don't actually have position data,
+      // but we enumerated all the locations earlier,
+      // so anytime one of these entities casts, we know where it is.
+      id: 'SophiaEX Tilt Via Cast',
+      netRegex: NetRegexes.startsUsing({ id: '19A9', source: 'Sophia' }),
+      netRegexDe: NetRegexes.startsUsing({ id: '19A9', source: 'Sophia' }),
+      netRegexFr: NetRegexes.startsUsing({ id: '19A9', source: 'Sophia' }),
+      netRegexJa: NetRegexes.startsUsing({ id: '19A9', source: 'ソフィア' }),
+      netRegexCn: NetRegexes.startsUsing({ id: '19A9', source: '索菲娅' }),
+      netRegexKo: NetRegexes.startsUsing({ id: '19A9', source: '소피아' }),
+      condition: function(data) {
+        // No platform tilts if clones are up, and no tilts if it's the first one.
+        return !data.clonesActive && data.scaleSophias;
+      },
+      // We let the storage triggers catch up before calling.
+      delaySeconds: .5,
+      suppressSeconds: 5,
+      alertText: function(data, matches) {
+        if (![2, 4, 6].includes(data.quasarTethers.length))
+          return;
+        let balance = 0;
+        // It's possible a 6-tether group could be 4/2, so we have to eliminate that possibility.
```

Comment:
That's fair. I feel like there was a reason behind why I did this, but I didn't write it down, so it can't have been that important. Maybe it was because there are *way* too many elements stored on `data` already and I was naively trying to reduce that? Regardless, that would definitely be a less verbose option.